common/v2/services/RatesProvider.tsx

```diff
@@ -1,54 +1,59 @@
-import React, { Component, createContext } from 'react';
+import React, { createContext, useContext, useState, useEffect } from 'react';
 
+import { StoreContext } from 'v2/services/Store';
 import { PollingService } from 'v2/workers';
-import { IRate } from 'v2/types';
+import { IRates, TTicker, StoreAsset } from 'v2/types';
 
 interface State {
-  rates: IRate[];
-  getRate(uuid: string): object | undefined;
-  getAllRates(): object[];
+  rates: IRates;
+  getRate(ticker: TTicker): number | undefined;
 }
 
+const DEFAULT_FIAT_PAIRS = ['USD', 'EUR'] as TTicker[];
+const DEFAULT_FIAT_RATE = 0;
+const POLLING_INTERRVAL = 60000;
+const RATES_URL = 'https://proxy.mycryptoapi.com/cc/multi';
+const buildQueryUrl = (assets: TTicker[], currencies: TTicker[]) => `
+  ${RATES_URL}/?fsyms=${assets.join(',')}&tsyms=${currencies.join(',')}
+`;
+
+// The cryptocompare api that our proxie uses fails gracefully and will return a conversion rate
+// even if some are tickers are invalid (e.g WETH, GoerliETH etc.)
+// @TODO: figure out how to handle the conversion more elegantly then `DEFAULT_FIAT_RATE`
+const getAssetTickers = (assets: StoreAsset[]): TTicker[] => assets.map(a => a.ticker as TTicker);
```

Comment:
You'll want to filter out duplicates here. Otherwise you get requests like this: ![image](https://user-images.githubusercontent.com/29407814/64274839-140bb180-cef9-11e9-9f5f-e28ae690b282.png)

Reply:
done 
by creating `assetTickers` attribute in StoreContext with `[...new Set(targetAssets.map(a => a.ticker as TTicker))]`